src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey
+<T extends Record<string, any>>(items: T[], key: string): GroupsMap<T> {
```

Comment:
You can set type directly without extenstion
Also you can set key as a key of the object T
```suggestion
<T>(items: T[], key: keyof T): GroupsMap<T> {
```